### PR TITLE
Unvendor nlohmann json

### LINKED ISSUE #14326: Unvendor json, and bring it from CPM
https://github.com/tenstorrent/tt-metal/tree/main/tt_metal/third_party/json

Bring this from CPM like all the other dependencies.

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
Closes #14326 

### Problem description
json is an external dependency, lets treat it like one.
By having it as an external dependency, we can easily:
- increment version
- Use library target in cmake
- Avoid bundling it in our wheel
- Avoid conflicts where multiple system components must align to the same json version
- Get it from the system if desired by customer

### What's changed
Use json like external dep.
Bring it in from CPM.
Use cmake library target to propagate include path.

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/12293242379)
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_common.hpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #pragma once
 
-#include "tt_metal/third_party/json/json.hpp"
+#include <nlohmann/json.hpp>
```
Comment (tt-rkim) [inline tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_common.hpp:7]: @patrickroberts @yan-zaretskiy 
So this is our preferred way of referring to external libraries that we pull in through any dependency management system (in this case, CPM)? So it's clear it's external?

**Review by tt-rkim — APPROVED** [review]

File: tests/ttnn/unit_tests/gtests/test_graph_query_op_constraints.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #include "impl/event/event.hpp"
 #include "impl/program/program.hpp"
 #include "tests/tt_metal/tt_metal/common/dispatch_fixture.hpp"
-#include "third_party/json/json.hpp"
+#include <nlohmann/json.hpp>
```
Comment (afuller-TT) [inline tests/ttnn/unit_tests/gtests/test_graph_query_op_constraints.cpp:21]: ❤️

File: tt_metal/third_party/json/json.hpp
Comment (afuller-TT) [inline tt_metal/third_party/json/json.hpp:1]: Wait this wasn't even a submodule?  We just copy-pasted the files??

**Review by afuller-TT — APPROVED** [review]

**Review by mo-tenstorrent — APPROVED** [review]

File: tt_metal/third_party/json/json.hpp
Comment (TT-billteng) [reply tt_metal/third_party/json/json.hpp:1]: lol

Comment (ayerofieiev-tt) [reply tt_metal/third_party/json/json.hpp:1]: 🤦‍♂️

File: tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_common.hpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_common.hpp:7]: external lib ideally you include via `<>`.
things from your project you include via `""`

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by TT-billteng — APPROVED** [review]

File: tt_metal/third_party/json/json.hpp
Comment (tt-rkim) [reply tt_metal/third_party/json/json.hpp:1]: That's what vendoring is hahaahahahahahaah
We are a funny "SW" engineering company

**Comment by tt-rkim** [comment]
POGGERS TY BOSS MAN

File: tt_metal/third_party/json/json.hpp
Comment (blozano-tt) [reply tt_metal/third_party/json/json.hpp:1]: I learned this term from Rayray

**Pull request merged**